functions.ts

```diff
@@ -0,0 +1,68 @@
+import AsyncStorage from '@react-native-community/async-storage';
+import _ from 'lodash';
+
+import { Gutkas, Settings } from './Config/defaults';
+import { storedGutka } from './Config/types';
+import { gutkaFetched, setttingsFetched } from './Config/interfaces'
+const GUTKAS_KEY = 'Gutkas';
+const SETTINGS_KEY = 'Settings';
+
+
+const fetchGutkas = async (currentGutkaName: string): Promise<gutkaFetched> => {
+  try {
+    let getGutkas = await AsyncStorage.getItem(`${GUTKAS_KEY}`);
+    if (getGutkas === null) {
+      await AsyncStorage.setItem(`${GUTKAS_KEY}`, JSON.stringify(Gutkas));
+      getGutkas = await AsyncStorage.getItem(`${GUTKAS_KEY}`);
+    }
+    const normalized = getGutkas !== null ? JSON.parse(getGutkas) : [];
+    let currGutka: string = currentGutkaName;
+    console.log(normalized.items);
+    if (currentGutkaName === "" || currentGutkaName === undefined) {
+      currGutka = normalized[0].name;
+    }
+    console.log(currentGutkaName);
+    const filter =
+      normalized.find((gutka: storedGutka) => (gutka.name === currGutka));
+    return {
+      $isDataReady: true,
+      $stored: normalized,
+      $currentName: currGutka,
+      $currentItems: _.values(filter.items)
+    }
+  } catch (e) {
+    throw new Error("Something went wrong and nothing could be fetched");
+  }
+}
+const fetchSettings = async (): Promise<setttingsFetched> => {
+  let getSettings = await AsyncStorage.getItem(`${SETTINGS_KEY}`);
+  if (getSettings === null) {
+    await AsyncStorage.setItem(`${SETTINGS_KEY}`, JSON.stringify(Settings));
+    getSettings = await AsyncStorage.getItem(`${SETTINGS_KEY}`);
+  }
+  const normalized = getSettings !== null ? JSON.parse(getSettings) : [];
+  const arr = _.values(normalized);
+  return {
+    $displayEngTransl: arr[0],
+    $displayPunTrasl: arr[1],
+    $displayTranslit: arr[2],
+    $gurmukhiSize: arr[3],
+    $translSize: arr[4],
+    $translitSize: arr[5]
+  }
+
+}
+
+const findCurrentGutka = async (gutkas: storedGutka[], name: string) => {
```

Comment:
No need of async here

Reply:
Oh i completely forgot about this 🤦‍♂ that would have solved so many problems from before. Thx bhai sahib ji!